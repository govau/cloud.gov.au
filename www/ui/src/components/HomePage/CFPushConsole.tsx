import * as React from "react";
import styled, { css } from "../../styled-components";

import { Console } from "../Window";

const Label = styled.span`
  font-weight: 600;
`;

const BinaryLabel = styled(Label)`
  ${({ theme }) =>
    css`
      color: ${theme.console.labels.binaryColor};
    `};
`;

const CFAppLabel = styled(Label)`
  ${({ theme }) =>
    css`
      color: ${theme.console.labels.cfAppColor};
    `};
`;

const OKLabel = styled(Label)`
  ${({ theme }) =>
    css`
      color: ${theme.console.labels.okColor};
    `};
`;

const CFAppStartedLabel = styled(Label)`
  ${({ theme }) =>
    css`
      color: ${theme.console.labels.cfAppStartedColor};
    `};
`;

const Prompt = styled.span`
  ${({ theme }) =>
    css`
      color: ${theme.console.promptColor};
    `};
`;

const Caret = styled.span`
  display: inline-block;
  width: 0.5rem;
  height: 1rem;
  vertical-align: middle;

  ${({ theme }) =>
    css`
      border-bottom: 1px solid ${theme.console.caretColor};
    `};
`;

const CFPushConsole: React.StatelessComponent = () => (
  <Console.Window>
    <Console.Header>bash</Console.Header>
    <Console.Body>
      ▲ <Prompt>~/my-app $</Prompt> ls
      <br />
      Procfile main.go manifest.yml <BinaryLabel>my-app</BinaryLabel>
      <br />▲ <Prompt>~/my-app $</Prompt> cf push
      <br />
      Uploading <CFAppLabel>my-app</CFAppLabel>...
      <br />
      Uploading 2.1M, 3 files
      <br />
      Done uploading
      <br />
      <OKLabel>OK</OKLabel>
      <br />
      1 of 1 instances running
      <br />
      <CFAppStartedLabel>App started</CFAppStartedLabel>
      <br />▲ <Prompt>~/my-app</Prompt> $ <Caret />
    </Console.Body>
  </Console.Window>
);

export default CFPushConsole;
