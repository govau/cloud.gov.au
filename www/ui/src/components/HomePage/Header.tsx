import * as React from "react";
import styled, { css } from "../../styled-components";
import { Flex, Box } from "grid-styled";

import Container from "../Container";

const Heading = styled.h1`
  margin-top: 0;
  margin-bottom: 0;
  font-size: 4rem;
  font-weight: 700;
`;

const StyledHeader = styled.div`
  ${({ theme }) =>
    css`
      background-color: ${theme.headerColor};
    `};
  color: #ffffff;
`;

interface Props {}

const Header: React.StatelessComponent<Props> = () => (
  <StyledHeader>
    <Container>
      <Flex wrap={true} pt={2} pb={3}>
        <Box>
          <Heading>Cloud</Heading>
        </Box>
      </Flex>
    </Container>
  </StyledHeader>
);

export default Header;
