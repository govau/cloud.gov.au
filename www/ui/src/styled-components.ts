import * as styledComponents from "styled-components";
import {
  ThemedStyledComponentsModule,
  StyledFunction
} from "styled-components";

import Theme from "./theme";

const {
  default: styled,
  css,
  injectGlobal,
  keyframes,
  ThemeProvider
} = (styledComponents as ThemedStyledComponentsModule<
  any
>) as ThemedStyledComponentsModule<Theme>;

export { css, injectGlobal, keyframes, ThemeProvider, StyledFunction };
export default styled;
