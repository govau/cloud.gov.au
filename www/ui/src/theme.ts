export default interface Theme {
  textColor: string;
  linkColor: string;
  linkUnderlineColor: string;
  linkHoverColor: string;
  linkHoverBgColor: string;
  bgColor: string;
  contentBgColor: string;
  heroColor: string;
  primaryColor: string;
  secondaryColor: string;
  backgroundColor: string;
  borderColor: string;
  disclaimerColor: string;
  headerColor: string;
  footerColor: string;
  footerTextColor: string;
  footerLogoDividerColor: string;
  deltaSignPositiveColor: string;
  deltaSignNegativeColor: string;
};

export const theme: Theme = {
  textColor: "#313131",
  linkColor: "inherit",
  linkUnderlineColor: "#00c3ef",
  linkHoverColor: "#313131",
  linkHoverBgColor: "#5bcbe3",
  bgColor: "#313131",
  contentBgColor: "#ffffff",
  heroColor: "#115361",
  primaryColor: "#1B7991",
  secondaryColor: "#313131",
  backgroundColor: "#f0f3f5",
  borderColor: "#dddddd",
  disclaimerColor: "#313131",
  headerColor: "#115361",
  footerColor: "#313131",
  footerTextColor: "#bdbdbd",
  footerLogoDividerColor: "#616161",
  deltaSignPositiveColor: "#0cac78",
  deltaSignNegativeColor: "#ff635c"
};
