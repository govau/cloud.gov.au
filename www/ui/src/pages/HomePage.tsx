import * as React from "react";
import Helmet from "react-helmet";
import styled, { css } from "../styled-components";
import { Flex, Box } from "grid-styled";

import Container from "../components/Container";
import Link from "../components/Link";
import Header from "../components/HomePage/Header";
import CFPushConsole from "../components/HomePage/CFPushConsole";
import DevEnvOne from "../components/DevEnvironment/DevEnvOne"
import DevEnvTwo from "../components/DevEnvironment/DevEnvTwo"

const body = "\n\n---\nSource: cloud.gov.au home page";

const Content = styled.div`
  ${({ theme }) =>
    css`
      border-top: 6px solid ${theme.headerDividerColor};
      background-color: ${theme.contentBgColor};
    `};
`;

const P = styled.p`
  font-size: 1.5rem;
`;

const Ribbon1 = styled.section`
  background: #fafafa;
  border-top: 1px solid;
  border-bottom: 1px solid;
  border-color: #eaeaea;
`;

const FeatureHeading = styled.h2`
  font-weight: 400;
  font-size: 1.5rem;
`;

const FeatureList = styled.ul`
  font-size: 1rem;
  line-height: 1.5rem;
`;

const FeatureP = styled(P)`
  font-size: 1rem;
  color: #666666;
`;

const HomePage: React.StatelessComponent = () => (
  <React.Fragment>
    <Helmet>
      <title />
    </Helmet>
    <Header />
    <Content>
      <Ribbon1>
        <Container px={0}>
          <Flex wrap={true} px={2} pt={[2, 4]} pb={4}>
            <Box w={[1, 1 / 2]} pt={[1, 3]} pb={[1, 3]} pr={[0, 3]}>
              <FeatureHeading>
                Focus on your <strong>services</strong>, not your servers
              </FeatureHeading>
              <FeatureP>
                Install{" "}
                <Link href="https://docs.cloudfoundry.org/cf-cli/">
                  Cloud Foundry CLI tools
                </Link>{" "}
                and deploy with ease
              </FeatureP>
              <FeatureList>
                <li>Write your code in any programming language</li>
                <li>Automatic HTTPS / certificates</li>
                <li>Deploy instantly, deploy often</li>
              </FeatureList>
            </Box>
            <Box w={[1, 1 / 2]} pt={[1, 3]} pb={[1, 3]} pl={[0, 3]}>
              <CFPushConsole />
            </Box>
          </Flex>
        </Container>
      </Ribbon1>

      <Container px={0}>
        <Flex wrap={true} px={2} pt={[0, 4]} pb={4}>
          <Box w={[1, 1 / 2]} pt={[1, 3]} pb={[1, 3]} pr={[0, 10]}>
            <DevEnvOne />
          </Box>
          <Box w={[1, 1 / 2]} pt={[2, 3]} pb={[1, 3]} pr={[0, 3]}>
            <DevEnvTwo />
          </Box>
        </Flex>
      </Container>

      <Container px={0}>
        <Flex wrap={true} px={2} py={[2, 4]}>
          <Box w={[1, 1 / 3]} pr={[0, 3]}>
            <FeatureHeading>Automatically secure</FeatureHeading>
            <FeatureP>
              We provision and renew SSL certificates automatically.
            </FeatureP>
            <FeatureP>
              Thanks to{" "}
              <Link href="https://letsencrypt.org/">Let’s Encrypt</Link>, you
              get security at zero cost, for as many domains as you want.
            </FeatureP>
          </Box>
          <Box w={[1, 1 / 3]} px={[0, 3]}>
            <FeatureHeading>Programming languages</FeatureHeading>
            <FeatureP>
              Supported languages and frameworks include Go, Java, .NET,
              Node.js, PHP, Python, Ruby, static files and binaries.
            </FeatureP>
            <FeatureP>
              Learn about{" "}
              <Link href="https://docs.cloudfoundry.org/buildpacks/">
                Cloud Foundry buildpacks
              </Link>
              .
            </FeatureP>
          </Box>
          <Box w={[1, 1 / 3]} pl={[0, 3]}>
            <FeatureHeading>Services</FeatureHeading>
            <FeatureP>
              Seamlessly connect your application to backend services.
            </FeatureP>
            <FeatureP>
              Available services include Postgres, MySQL and ElasticSearch.
            </FeatureP>
            <FeatureP>
              <Link
                href={`mailto:support@cloud.gov.au?subject=${encodeURIComponent(
                  "Enquiry about cloud.gov.au - interested in missing service"
                )}&body=${encodeURIComponent(body)}`}
              >
                Get in contact with us
              </Link>{" "}
              if we’re missing a service that you need.
            </FeatureP>
          </Box>
        </Flex>
      </Container>
    </Content>
  </React.Fragment>
);

export default HomePage;
