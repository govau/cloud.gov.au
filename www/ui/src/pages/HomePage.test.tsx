import * as React from "react";
import { MemoryRouter } from "react-router";
import * as renderer from "react-test-renderer";

import HomePage from "./HomePage";
import { theme } from "../theme";
import { ThemeProvider } from "styled-components";

it("renders without crashing", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <ThemeProvider theme={theme}>
          <HomePage />
        </ThemeProvider>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
