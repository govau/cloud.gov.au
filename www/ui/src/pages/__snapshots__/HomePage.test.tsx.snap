// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div
    className="sc-htoDjs dFvRPE"
>
    <div
        className="sc-bxivhb dyGNMJ sc-bdVaJa bNqJOV"
        is={null}
    >
        <div
            className="sc-htpNat jtWIOA sc-bdVaJa eSFxna"
            is={null}
        >
            <div
                className="sc-bdVaJa iHZvIS"
                is={null}
            >
                <h1
                    className="sc-bZQynM bwMZIe"
                >
                    Cloud
                </h1>
                <p
                    className="sc-gzVnrw lgDsWe"
                >
                    A platform to make delivering and operating digital services easier.
                </p>
            </div>
        </div>
    </div>
</div>,
  <div
    className="sc-dnqmqq jNPqvJ"
>
    <div
        className="sc-bxivhb dyGNMJ sc-bdVaJa bNqJOV"
        is={null}
    >
        <div
            className="sc-htpNat jtWIOA sc-bdVaJa iHZvIS"
            is={null}
        >
            <div
                className="sc-bdVaJa jFKWis"
                is={null}
            >
                <p
                    className="sc-iwsKbI eEZkcX"
                >
                    <a
                        className="sc-EHOje dByega"
                        href="/"
                        onClick={[Function]}
                    >
                        <strong>
                            cloud.gov.au
                        </strong>
                    </a>
                     
                    is a faster, safer, standard way to change web apps without impacting users. It makes it easier for government to release, monitor and grow user-facing digital services, freeing up teams to focus on writing code that meets user needs.
                </p>
                <p
                    className="sc-iwsKbI eEZkcX"
                >
                    As of January 2018, cloud.gov.au has 27 apps in production, and 86 apps in development across our programs of work.
                </p>
                <p
                    className="sc-iwsKbI eEZkcX"
                >
                    DTA services running on cloud.gov.au include the
                     
                    <a
                        className="sc-ifAKCX cLeGDS"
                        href="https://www.dta.gov.au/what-we-do/platforms/marketplace/"
                    >
                        Digital Marketplace
                    </a>
                    , the
                     
                    <a
                        className="sc-ifAKCX cLeGDS"
                        href="https://www.dta.gov.au/what-we-do/platforms/performance/"
                    >
                        Performance Dashboard
                    </a>
                    , the
                     
                    <a
                        className="sc-ifAKCX cLeGDS"
                        href="https://smart-cities.dashboard.gov.au/"
                    >
                        National Cities Performance Framework Dashboard
                    </a>
                    , and the australia.gov.au
                     
                    <a
                        className="sc-ifAKCX cLeGDS"
                        href="https://media.australia.gov.au"
                    >
                        Media Release Service
                    </a>
                    .
                </p>
                <p
                    className="sc-iwsKbI eEZkcX"
                >
                    We
                     
                    <a
                        className="sc-ifAKCX cLeGDS"
                        href="http://status.cloud.gov.au"
                    >
                        publish real time performance statistics
                    </a>
                     
                    of both the cloud.gov.au platform itself, and the apps running on cloud.gov.au.
                </p>
                <p
                    className="sc-iwsKbI eEZkcX"
                >
                    If you are interested in building new digital services on cloud.gov.au, please
                     
                    <a
                        className="sc-ifAKCX cLeGDS"
                        href="mailto:support@cloud.gov.au?subject=Enquiry%20about%20cloud.gov.au&body=%0A%0A---%0ASource%3A%20cloud.gov.au%20home%20page"
                    >
                        get in contact with us
                    </a>
                    .
                </p>
            </div>
        </div>
    </div>
</div>,
]
`;
